src/CodeGenerator/src/Generator/RequestSerializer/RestJsonSerializer.php

```diff
@@ -172,17 +169,29 @@ private function dumpArrayList(string $output, string $input, string $contextPro
 
         return strtr('
             $index = -1;
+            INIT_CODE
             foreach (INPUT as $listValue) {
                 $index++;
                 MEMBER_CODE
             }
         ',
             [
                 'INPUT' => $input,
+                'INIT_CODE' => $this->dumpArrayInit($output, $memberShape),
                 'MEMBER_CODE' => $memberCode = $this->dumpArrayElement(sprintf('%s[$index]', $output), '$listValue', $contextProperty, $memberShape, true),
             ]);
     }
 
+    private function dumpArrayInit(string $output, Shape $shape)
+    {
+        switch (true) {
+            case $shape instanceof StructureShape:
+                return strtr('$payloadOUTPUT = [];', [
```

Comment:
I style don't get why we initialize the array when children is a structure, and we don't do it for map, list, scalar...

Reply:
After latest change, it could be useless now, I will give it a try !

Reply:
Ok good catch, I just removed this stuff